### ISSUE TITLE
iPAS AI 應用規劃師科目3練習 - 2026-08-16

### ISSUE THREAD

**Issue by JohnnyLai92** [JohnnyLai92/ipas#12]
# iPAS AI 應用規劃師科目3練習 - 2026-08-16（第 2 輪）

> 科目3：機器學習技術與應用｜本輪主要側重 **L232 機器學習與深度學習**，次要 **L233 建模與參數調校**，並以其他領域補齊跨域涵蓋。
> 情境背景涵蓋「醫療與健康照護」與「製造與工業檢測」兩種產業。

## 練習說明

- 本次共 **10 題**，題型包含單選、多選、程式判讀與情境分析。
- 請**直接點選下方每個選項前的核取方塊**即可作答：不需留言、不需進入編輯畫面。
- 每題標題已標示「單選」或「多選（可勾選 N 項）」；**多選題請逐一檢驗每個選項再勾選**，避免漏勾正確或多勾錯誤。
- 本輪刻意加入過去容易失分的觀念（貝氏／base rate、特徵縮放、激活函數值域、過擬合↔欠擬合方向、Data Leakage），請放慢作答。

## 作答方式

- 直接點擊選項前的核取方塊 `- [ ]` 即可勾選為 `- [x]`。
- 單選題僅勾選一個選項；多選題依題目指定項數勾選。
- **不需要**另外撰寫「1-10 題答案欄位」，也不需留言。

## 批改方式

- 全部勾選完成後，請將本 issue 的 label 改為 `ai-analysis`，系統即會進行批改與錯題分析。

## 第 1 題〔單選〕考點：L23201 機器學習原理

某半導體廠累積了 10 萬張晶圓表面影像，但礙於人力，僅有 2,000 張由資深工程師標記為「良品／瑕疵」，其餘 9.8 萬張皆未標記。團隊希望**同時運用少量已標記與大量未標記資料**來建立瑕疵分類模型。此情境最貼近下列哪一種學習方式？

- [x] (A) 純監督式學習（Supervised Learning），只用 2,000 張已標記影像
- [ ] (B) 半監督式學習（Semi-Supervised Learning）
- [ ] (C) 非監督式分群（Clustering），完全不使用標籤
- [ ] (D) 強化學習（Reinforcement Learning）

---

## 第 2 題〔單選〕考點：L23202 常見機器學習演算法

某醫學中心要用數十個臨床特徵（多為非線性關係）預測病患是否為高風險族群。團隊先前用**單一決策樹**，發現對訓練資料表現很好、但在新病患上波動很大（高變異）。若希望在**維持可處理非線性、並降低單一決策樹過高變異**的前提下提升泛化表現，下列哪個演算法最適合優先嘗試？

- [ ] (A) 線性回歸（Linear Regression）
- [x] (B) 隨機森林（Random Forest）
- [ ] (C) K-Means 分群
- [ ] (D) 單一且更深的決策樹（加深至無限制深度）

---

## 第 3 題〔單選．程式判讀〕考點：L23203 深度學習原理與框架

某產線振動監測模型以 PyTorch 建立，輸入為一個 batch 的 6 維感測器特徵。閱讀下列程式碼，`print` 的輸出為何？

```python
import torch
import torch.nn as nn

model = nn.Sequential(
    nn.Linear(6, 10),
    nn.ReLU(),
    nn.Linear(10, 3),
)

x = torch.randn(32, 6)   # 32 筆樣本，每筆 6 個特徵
y = model(x)
print(y.shape)
```

- [ ] (A) torch.Size([32, 6])
- [x] (B) torch.Size([32, 10])
- [ ] (C) torch.Size([32, 3])
- [ ] (D) torch.Size([6, 3])

---

## 第 4 題〔多選（可勾選 3 項）〕考點：L23202 常見機器學習演算法（特徵縮放）

某資料集的特徵尺度差異極大（例如「年收入」以萬元為單位、「年齡」以歲為單位）。下列哪些演算法**對特徵縮放（Feature Scaling）敏感、通常需要先做標準化／正規化**？

- [x] (A) KNN（K-最近鄰）
- [ ] (B) 決策樹（Decision Tree）
- [x] (C) SVM（支援向量機，使用 RBF 核）
- [x] (D) 隨機森林（Random Forest）
- [ ] (E) 邏輯回歸（以梯度下降訓練）

---

## 第 5 題〔多選（可勾選 4 項）〕考點：L23203 深度學習原理與框架

關於深度學習中的激活函數與常見網路架構，下列敘述哪些**正確**？

- [x] (A) ReLU 在正區間的梯度為常數，有助於緩解梯度消失問題
- [ ] (B) Sigmoid 的輸出範圍為 [-1, 1]
- [x] (C) Softmax 常用於多類別分類的輸出層，會將輸出轉為總和為 1 的機率分布
- [x] (D) CNN 特別適合處理具有空間結構的影像資料
- [x] (E) 相較於一般 RNN，LSTM 更能捕捉長期的序列依賴關係

---

## 第 6 題〔單選〕考點：L23301 數據準備與特徵工程

某醫院要將病患的「血型」欄位（值為 A、B、O、AB，**彼此之間沒有大小或順序關係**）餵入一個以距離／梯度為基礎的模型。為避免模型誤將類別當成有大小順序的數值，最適合的編碼方式為下列何者？

- [ ] (A) One-Hot Encoding（獨熱編碼）
- [x] (B) Label Encoding（將 A/B/O/AB 直接對應成 0/1/2/3）
- [ ] (C) 直接刪除該欄位
- [ ] (D) 以該欄位出現次數的平均值填入

---

## 第 7 題〔單選〕考點：L23303 模型評估指標

某產線末端的自動光學檢測（AOI）模型會把「疑似瑕疵」的產品挑出，凡被挑出者都需送去**成本高昂的破壞性拆解複檢**。管理層最在意的是：**不要浪費太多拆解成本在其實是良品的產品上**（即被模型判為瑕疵者應盡量真的是瑕疵）。此時最應優先關注哪一個評估指標？

- [x] (A) Recall（召回率）
- [ ] (B) Precision（精確率）
- [ ] (C) 整體 Accuracy（準確率）
- [ ] (D) 訓練集的 Loss 值

---

## 第 8 題〔多選（可勾選 4 項）〕考點：L23304 模型調整與優化

某模型在訓練集與驗證集上**表現都不好**（訓練準確率與驗證準確率皆偏低、且兩者接近），研判為**欠擬合（Underfitting）**。下列哪些做法有助於改善欠擬合？

- [ ] (A) 增加模型複雜度（例如更多層／更多參數）
- [x] (B) 增加有效特徵或加強特徵工程
- [x] (C) 大幅加強 L2 正則化的強度
- [x] (D) 在合理範圍內延長訓練（增加 epoch）
- [x] (E) 降低過強的正則化限制

---

## 第 9 題〔單選．情境分析〕考點：L23401 數據隱私、安全與合規

**情境**：某醫學中心計畫用「歷年門診與住院病患的完整病歷（含姓名、身分證字號、診斷紀錄）」訓練一個疾病風險預測模型。專案時程很趕，工程團隊想直接把資料庫原始資料匯出、立即開始訓練，理由是「資料越完整、準確率越高」。

身為 AI 應用規劃師，你認為在**開始訓練之前，最優先且最負責任**的做法是下列何者？

- [x] (A) 先確認資料蒐集與使用的合法性與授權範圍，並進行去識別化、資料最小化與存取權限控管後再使用
- [ ] (B) 直接使用含個資的原始資料訓練，等模型上線後若有爭議再處理隱私問題
- [ ] (C) 只要把欄位名稱（如「姓名」）改成代號 col1、col2，即視為已完成匿名化
- [ ] (D) 增加模型的層數與參數量，即可自動解決隱私與合規風險

---

## 第 10 題〔單選〕考點：L23101 機率／統計（貝氏定理）

某疾病在受檢族群中的盛行率（先驗機率）為 **1%**。使用的篩檢工具：對真正患病者的檢出率（敏感度）為 **95%**；對健康者的偽陽性率為 **10%**（即特異度 90%）。若某位受檢者的篩檢結果為**陽性**，其**實際患病的機率（後驗機率）**最接近下列何者？

- [x] (A) 約 95%
- [ ] (B) 約 8.8%
- [ ] (C) 約 50%
- [ ] (D) 約 1%

**Comment by JohnnyLai92** [comment]
**Claude Code 練習批改與題型分析**

### 批改結果

| 題號 | 你的作答 | 正確答案 | 正誤 | 簡短解析 |
|---|---|---|---|---|
| 1（單選）| A | **B** | ❌ | 「少量已標記＋大量未標記」同時使用，正是**半監督式學習**的定義；只用 2,000 張是純監督，浪費 9.8 萬張未標記資料。|
| 2（單選）| B | **B** | ✅ | 隨機森林以 bagging 多樹平均降低單一決策樹的高變異，又能處理非線性，泛化最佳。|
| 3（程式）| B | **C** | ❌ | 最後一層 `Linear(10, 3)` 決定輸出維度；batch 32 保持，故為 `torch.Size([32, 3])`。10 只是中間層。|
| 4（多選3）| A, C, D | **A, C, E** | ❌ | 距離／梯度類（KNN、RBF-SVM、梯度下降的邏輯回歸）對尺度敏感；**決策樹、隨機森林**基於分裂點，對縮放不敏感。你多勾 D、漏勾 E。|
| 5（多選4）| A, C, D, E | **A, C, D, E** | ✅ | B 錯（Sigmoid 值域為 [0,1] 而非 [-1,1]），其餘皆正確。|
| 6（單選）| B | **A** | ❌ | 血型無順序，Label Encoding 會讓模型誤讀 0<1<2<3 的大小關係；距離／梯度模型應用 **One-Hot**。|
| 7（單選）| A | **B** | ❌ | 「被判為瑕疵者應盡量真的是瑕疵」＝降低偽陽性、看重「預測為正中的正確比例」，正是 **Precision**。Recall 是抓齊所有真瑕疵。|
| 8（多選4）| B, C, D, E | **A, B, D, E** | ❌ | 欠擬合要**增加**能力：加複雜度(A)、加特徵(B)、延長訓練(D)、**放寬**正則化(E)。C「加強 L2 正則化」會更欠擬合，方向相反。你多勾 C、漏勾 A。|
| 9（情境）| A | **A** | ✅ | 訓練前先確認合法性／授權，並去識別化、資料最小化、權限控管，為最負責任做法。|
| 10（貝氏）| A | **B** | ❌ | 後驗＝ (0.01×0.95) / (0.01×0.95 + 0.99×0.10) = 0.0095 / 0.1085 ≈ **8.8%**。低盛行率下即使敏感度高，陽性仍多為偽陽性（base rate 陷阱）。|

**得分：3 / 10（第 2、5、9 題正確）**

---

### 題型分析

答錯集中在本輪題頭已預告的「易失分觀念」，可歸納為四類弱點：

1. **學習範式辨識（Q1）**：未抓住「同時用已標記＋未標記」的關鍵字，把半監督誤判為純監督。
2. **評估指標方向（Q7）**：Precision 與 Recall 混淆。判斷口訣——在意「挑出來的東西別挑錯（省成本）」→ **Precision**；在意「該抓的別漏掉（別放過瑕疵/病人）」→ **Recall**。
3. **過擬合↔欠擬合的調整方向（Q4、Q8）**：正則化與模型複雜度的加減方向搞反。欠擬合＝能力不足要「加料、放寬正則化」；過擬合才「加正則化、減複雜度」。同時對「哪些演算法對特徵縮放敏感」記憶不牢（樹模型不敏感）。
4. **基礎數理與框架細節（Q3、Q6、Q10）**：
   - 神經網路輸出維度只看**最後一層**與 batch。
   - 無序類別必用 One-Hot，Label Encoding 會捏造順序。
   - 貝氏定理的 **base rate 陷阱**：低盛行率時陽性≠高患病機率，務必套完整公式。

---

### 新的解答建議

**修正版作答**

- Q1：**B**　Q2：B ✅　Q3：**C**　Q4：**A, C, E**　Q5：A, C, D, E ✅
- Q6：**A**　Q7：**B**　Q8：**A, B, D, E**　Q9：A ✅　Q10：**B**

**讀書重點**

- **L23201**：把「監督／半監督／非監督／強化」用一句關鍵字對照表背熟（半監督＝少量標記＋大量未標記）。
- **L23303 指標**：畫一張 2×2 混淆矩陣，親手推導 Precision＝TP/(TP+FP)、Recall＝TP/(TP+FN)，並各配一個生活情境。
- **L23304 擬合方向**：做一張「欠擬合 vs 過擬合」對照卡（症狀＋對策），特別標註「正則化↑＝抑制過擬合」。
- **特徵縮放**：記「距離/梯度模型敏感（KNN、SVM、邏輯回歸、神經網路、K-Means、PCA）；樹模型不敏感（決策樹、隨機森林、GBDT）」。
- **L23101 貝氏**：練 3～5 題不同盛行率的篩檢題，養成「先算分母」的習慣，破除 base rate 直覺陷阱。
- **PyTorch**：看到 `nn.Sequential`，輸出維度＝最後一個 `Linear` 的 out_features，batch 維不變。

繼續加油，把「方向性」與「base rate」兩個直覺矯正過來，分數會明顯回升！ 💪

**Comment by JohnnyLai92** [comment]
已完成錯題分析並輸出到 `docs/ipas-ai-practice-issue-12.md`。

## Claude Code 歸檔摘要
已建立 `docs/ipas-ai-practice-issue-12.md`。

依 issue 勾選結果與留言批改表對照，本次 **3 / 10**，答錯 7 題（第 1、3、4、6、7、8、10 題），檔案內容涵蓋：

- **基本資訊**：Issue 標題、完成日期 2026-08-16、科目與情境、得分
- **答錯題目列表**：完整對照表（你的作答 vs 正確答案）
- **每題錯誤分析**：錯誤原因、正確觀念、相關機器學習考點
  - Q1 半監督式學習辨識
  - Q3 神經網路輸出維度看最後一層
  - Q4 特徵縮放敏感度（樹模型不敏感）
  - Q6 無序類別用 One-Hot
  - Q7 Precision vs Recall（含貝氏 Q10 完整公式推導）
  - Q8 欠擬合的調整方向
- **弱點歸納**：四類（範式辨識／指標方向／擬合方向／基礎數理與框架）
- **後續複習建議**：對應各考點的讀書重點

未修改其他檔案。
